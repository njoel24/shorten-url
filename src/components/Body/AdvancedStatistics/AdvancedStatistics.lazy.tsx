import React, { lazy, Suspense } from 'react';

const LazyAdvancedStatistics = lazy(() => import('./AdvancedStatistics'));

const AdvancedStatistics = (props: JSX.IntrinsicAttributes & { children?: React.ReactNode; }) => (
  <Suspense fallback={null}>
    <LazyAdvancedStatistics {...props} />
  </Suspense>
);

export default AdvancedStatistics;
