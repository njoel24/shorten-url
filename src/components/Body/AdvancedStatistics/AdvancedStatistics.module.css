.AdvancedStatistics {
    color: var(--veryDarkBlue);
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 500px;
    text-align: center;
    margin-top: 40px;
}

.AdvancedStatistics .ContentWrapper {
    margin-bottom: 40px;
}

.AdvancedStatistics h1 {
    font-family: 'Poppins-Bold';
    margin-bottom: 0;
    font-size: 2.5rem;
}

.AdvancedStatistics h6 {
    font-family: 'Poppins-Light';
    color: var(--grayishViolet);
    margin-top: 20px;
    font-size: 1.1rem;
}