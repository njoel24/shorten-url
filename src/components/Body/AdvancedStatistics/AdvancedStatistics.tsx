import React from 'react';
import styles from './AdvancedStatistics.module.css';
import InfoBlockWrapper from './InfoBlockWrapper/InfoBlockWrapper';

const AdvancedStatistics = () => (
  <div className={styles.AdvancedStatistics} data-testid="AdvancedStatistics">
    <div className={styles.ContentWrapper}>
      <h1>Advanced Statistics</h1>
      <h6>Track how your links are performing across the web with our advanced statitics dashboard</h6>
    </div>
    <InfoBlockWrapper />
  </div>
);

export default AdvancedStatistics;
