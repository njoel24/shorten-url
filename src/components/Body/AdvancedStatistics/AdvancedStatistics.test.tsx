import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AdvancedStatistics from './AdvancedStatistics';

describe('<AdvancedStatistics />', () => {
  test('it should mount', () => {
    render(<AdvancedStatistics />);
    
    const advancedStatistics = screen.getByTestId('AdvancedStatistics');

    expect(advancedStatistics).toBeInTheDocument();
  });

  test('it should match the snapshot', () => {
    const wrapper = render(<AdvancedStatistics />);

    expect(wrapper).toMatchSnapshot();
  });
});