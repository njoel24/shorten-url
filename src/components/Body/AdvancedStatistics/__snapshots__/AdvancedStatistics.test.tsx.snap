// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedStatistics /> it should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="AdvancedStatistics"
        data-testid="AdvancedStatistics"
      >
        <div
          class="ContentWrapper"
        >
          <h1>
            Advanced Statistics
          </h1>
          <h6>
            Track how your links are performing across the web with our advanced statitics dashboard
          </h6>
        </div>
        <div
          class="InfoBlockWrapper"
          data-testid="InfoBlockWrapper"
        >
          <div
            class="InfoBlock"
            data-testid="InfoBlock"
          >
            <div
              class="IconWrapper"
            >
              <svg>
                icon-brand-recognition.svg
              </svg>
            </div>
            <div
              class="ContentWrapper"
            >
              <h3>
                Brand Recognition
              </h3>
              <h6>
                Boost your brand recognition with each click. Generic links don't mean a thing. Branded links help instil confidence in your content.
              </h6>
            </div>
          </div>
          <div
            class="LinkBlock"
          />
          <div
            class="InfoBlock"
            data-testid="InfoBlock"
          >
            <div
              class="IconWrapper"
            >
              <svg>
                icon-detailed-records.svg
              </svg>
            </div>
            <div
              class="ContentWrapper"
            >
              <h3>
                Detailed Records
              </h3>
              <h6>
                Gain insights who is clicking your links. Knowing when and where people engage with your content  helps inform better decisions
              </h6>
            </div>
          </div>
          <div
            class="LinkBlock"
          />
          <div
            class="InfoBlock"
            data-testid="InfoBlock"
          >
            <div
              class="IconWrapper"
            >
              <svg>
                icon-fully-customizable.svg
              </svg>
            </div>
            <div
              class="ContentWrapper"
            >
              <h3>
                Fully Customizable
              </h3>
              <h6>
                Improve brand awareness and content discoverability through customizable links, supercharging audience engagement
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="AdvancedStatistics"
      data-testid="AdvancedStatistics"
    >
      <div
        class="ContentWrapper"
      >
        <h1>
          Advanced Statistics
        </h1>
        <h6>
          Track how your links are performing across the web with our advanced statitics dashboard
        </h6>
      </div>
      <div
        class="InfoBlockWrapper"
        data-testid="InfoBlockWrapper"
      >
        <div
          class="InfoBlock"
          data-testid="InfoBlock"
        >
          <div
            class="IconWrapper"
          >
            <svg>
              icon-brand-recognition.svg
            </svg>
          </div>
          <div
            class="ContentWrapper"
          >
            <h3>
              Brand Recognition
            </h3>
            <h6>
              Boost your brand recognition with each click. Generic links don't mean a thing. Branded links help instil confidence in your content.
            </h6>
          </div>
        </div>
        <div
          class="LinkBlock"
        />
        <div
          class="InfoBlock"
          data-testid="InfoBlock"
        >
          <div
            class="IconWrapper"
          >
            <svg>
              icon-detailed-records.svg
            </svg>
          </div>
          <div
            class="ContentWrapper"
          >
            <h3>
              Detailed Records
            </h3>
            <h6>
              Gain insights who is clicking your links. Knowing when and where people engage with your content  helps inform better decisions
            </h6>
          </div>
        </div>
        <div
          class="LinkBlock"
        />
        <div
          class="InfoBlock"
          data-testid="InfoBlock"
        >
          <div
            class="IconWrapper"
          >
            <svg>
              icon-fully-customizable.svg
            </svg>
          </div>
          <div
            class="ContentWrapper"
          >
            <h3>
              Fully Customizable
            </h3>
            <h6>
              Improve brand awareness and content discoverability through customizable links, supercharging audience engagement
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
